packages/composites/token-allocation/src/adapter.ts

```diff
@@ -1,165 +1,37 @@
-    const decimals = _toValidDecimals(t.decimals)
-    return {
-      symbol: _toValidSymbol(t.symbol),
-      decimals,
-      balance: _toValidBalance(t.balance, decimals),
-    }
-  })
-}
-
-const computePrice = async (
-  source: string,
-  getPrices: GetPrices,
-  allocations: TokenAllocations,
-  quote: string,
-) => {
-  const symbols = (allocations as TokenAllocations).map((t) => t.symbol)
-  const payload = await getPrices(symbols, quote)
-  const result = priceTotalValue(source, allocations, quote, payload)
-  return { payload, result }
-}
-
-const computeMarketCap = async (
-  source: string,
-  getPrices: GetPrices,
-  allocations: TokenAllocations,
-  quote: string,
-) => {
-  const symbols = (allocations as TokenAllocations).map((t) => t.symbol)
-  const payload = await getPrices(symbols, quote, true)
-
-  const result = marketCapTotalValue(source, allocations, quote, payload)
-  return { payload, result }
-}
-
-const inputParams = {
-  source: false,
-  allocations: true,
-  quote: false,
-  method: false,
-}
-
-export const execute = async (input: AdapterRequest, config: Config): Promise<AdapterResponse> => {
-  const paramOptions = makeOptions(config)
-  const validator = new Validator(input, inputParams, paramOptions)
-  if (validator.error) throw validator.error
-
-  const jobRunID = validator.validated.id
-  const { quote = config.defaultQuote, method = config.defaultMethod } = validator.validated.data
-  const allocations = toValidAllocations(validator.validated.data.allocations)
-  const source: string = (
-    validator.validated.data.source ||
-    config.defaultSource ||
-    ''
-  ).toLowerCase()
-  if (source === '') {
-    throw Error('No source specified in the request or config!')
-  }
-
-  const sourceConfig = config.sources[source]
-
-  const _success = (payload: ResponsePayload, result: number) =>
-    Requester.success(
-      jobRunID,
-      {
-        status: 200,
-        data: { sources: [], payload, result },
-      },
+import { Builder } from '@chainlink/ea-bootstrap'
+import { AdapterRequest, APIEndpoint, ConfigFactory } from '@chainlink/types'
+import { adapters, makeConfig as makeUpstreamConfig } from './config'
+import { Config } from './types'
+
+const endpointMap = Object.fromEntries(adapters.map((adapter) => [adapter.NAME, adapter.endpoints]))
+
+/**
+ * endPointSelector Factory that either:
+ * - accepts a static upstream endpoint name
+ *
+ * OR
+ *
+ * - dynamically uses "method" as the endpoint
+ */
+
+export const makeEndpointSelector = <C>(
+  makeDownstreamConfig: ConfigFactory<C>,
+  downstreamEndpoints: Record<string, APIEndpoint<C>>,
+  upstreamEndpointName?: string,
+): ((request: AdapterRequest) => APIEndpoint<C>) => {
+  const upstreamConfig = makeUpstreamConfig()
+  return (request: AdapterRequest): APIEndpoint<C> => {
+    const endpointName = !upstreamEndpointName
+      ? request.data['method'] || upstreamConfig.defaultMethod
+      : upstreamEndpointName
```

Comment:
I think this also works
```suggestion
    const endpointName = upstreamEndpointName ?? request.data['method'] ?? upstreamConfig.defaultMethod
```